oracle/datadog_checks/oracle/oracle.py

```diff
@@ -156,6 +157,7 @@ def _connection(self):
                     self.log.error("The JDBC connection failed with the following error: %s", str(e))
                     self._connection_errors += 1
             else:
+                self.can_use_instant_client()
```

Comment:
I would rename `can_use_instant_client()` to `attempt_init_instant_client` since the function isn't returning a True or False on if instance client can be used, rather it is trying to initialize the Oracle client. 

I would also change this logic to
```
if self._use_instant_client:
    self.attempt_init_instant_client()
...
```

Reply:
I just added the content to the `_connection` function definition 🙇